### PR TITLE
Shivani | Test | Fetch Go cardless integrated bank balance in dashboard page

### PR DESCRIPTION
**Pull request by shivanivani02**
https://app.clickup.com/t/86cx3fz91

## Summary by CodeRabbit

- **New Features**
	- Introduced a new property to manage unlinked bank accounts.
	- Added functionality to link a single bank account directly when there is only one unlinked account.
	- Implemented a method for streamlined linking of bank accounts.

- **Bug Fixes**
	- Enhanced logic for managing linked and unlinked bank accounts to improve transaction handling.

### CODE REVIEW

**Review by ankitdubeyoffice — APPROVED** [review]

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 1]
```diff
@@ -3195,16 +3210,35 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
-            bankList: this.selectedBankUniqueName,
-            accountUniqueName: this.lc.accountUnq,
-        }
-        const dialogRef = this.dialog.open(BankLinkComponent, {
-            data: data,
-            panelClass: ['mat-dialog-md'],
-            disableClose: true
-        });
+        if (this.singleBank.length === 1) {
+            this.linkBankAccount()
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
            this.linkBankAccount();
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 2]
```diff
@@ -3195,16 +3210,35 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
+        } else {
+            const data = {
+                bankList: this.selectedBankUniqueName,
+                accountUniqueName: this.lc.accountUnq,
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
                accountUniqueName: this.lc.accountUnq
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 3]
```diff
@@ -1030,6 +1040,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
                     this.isBankAccountConnected = true;
                 } else {
                     this.showBankLinkButton = response.body.some(bank => Object.keys(bank.account).length === 0);
+                    this.singleBank = response.body.filter(bank => Object.keys(bank.account).length === 0);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
                    this.unlinkBankList = response.body.filter(bank => Object.keys(bank.account).length === 0);
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**